scripts/chatty_news.js

```diff
@@ -0,0 +1,512 @@
+                    localStorage.setItem(chattynews.LOCALSTORAGE_LASTIMAGE,imgurl); //only save image url if we actually got one.
+                }else{
+                    imgurl = "http://shackwiki.com/images/a/a8/IconBig.png"; //backup image if an error occurred - Shack Chatty Crest.
+                }
+                
+                chattynews.showImage(imgurl);
+            },
+            
+            showImage: function(imgurl){
+                //put the image in the background of the chatty
+                var imgdiv = document.createElement("div");
+                var maskimg = document.createElement("div");
+                imgdiv.setAttribute('id', "SNCNbackimg");
+                imgdiv.setAttribute('style','background-image: url("' + imgurl + '")');
+                maskimg.setAttribute('id', "SNCNmaskimg");
+                //maskimg.setAttribute('style', 'background-color: transparent !important;');
+                imgdiv.appendChild(maskimg);
+                document.body.insertBefore(imgdiv,document.body.firstChild);
+                
+                
+                console.log("ShacknewsChattyNews - image done "+((new Date()).getTime() - chattynews.scriptStartTime) + "ms");
+                
+                
+                if(getSetting("chatty_news_brighten_image")){
+                    var scroll = function(arg){
+                        var body = document.body,
+                            html = document.documentElement;
+
+                        var height = Math.max( body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight );
+                        var h = Math.max(document.documentElement.clientHeight, window.innerHeight || 0)
+                        var ele = document.getElementById('SNCNmaskimg');
+                        if( (window.pageYOffset<250) || (window.pageYOffset > (height - (h*2) ))){
+                            if(chattynews.tintback){
+                                chattynews.tintback = false;
+                                ele.setAttribute('style','background-color: transparent !important;');
+                            }
+                        }else if(!chattynews.tintback){
+                            chattynews.tintback = true;
+                            ele.setAttribute('style','');
+                        }
+                    }
+                    
+                    document.addEventListener("scroll", scroll);
+                    scroll();
+                }
+            },
+            
+            
+            
+            valueFromTagName: function (item, tagname) {
+                var val = item.getElementsByTagName(tagname);
+                if(val.length>0 && val[0]!=null && val[0].firstChild!=null)
+                    return val[0].firstChild.nodeValue;
+                return "";
+            },
+            
+            html_entity_decode: function (string, quote_style) {
```

Comment:
This whole HTML entity decoding business can likely be done with the following snippet:

```
function decodeHtml(html) {
// see: https://stackoverflow.com/a/42182294
    var txt = document.createElement("textarea");
    txt.innerHTML = html;
    return txt.value;
}
```

With some more cleverness you can also grab just innerText without tags, and it would be significantly more readable and easier to test than this chunk of code.